src/sentry/models/release.py

```diff
@@ -92,8 +92,56 @@ def _on_post(project, trigger):
         ):
             schedule_invalidate_project_config(project_id=project.id, trigger=trigger)
 
+    def _subscribe_project_to_alert_rule(project, release, trigger):
+        """
+        TODO: potentially enable custom query_extra to be passed on ReleaseProject creation (on release/deploy)
+        """
+        from django.utils import timezone
+
+        from sentry.incidents.logic import subscribe_projects_to_alert_rule
+        from sentry.incidents.models import (
+            AlertRule,
+            AlertRuleActivationConditionType,
+            AlertRuleMonitorType,
+        )
+
+        try:
+            project_alert_rules = AlertRule.filter(project=project)
+            if project_alert_rules.exists():
+                # If we have an AlertRule for the project
+                for par in project_alert_rules:
+                    if (
+                        par.monitor_type == AlertRuleMonitorType.ACTIVATED
+                        and par.activation_conditions
+                        == AlertRuleActivationConditionType.RELEASE_CREATION
+                    ):
+                        query_extra = (
+                            f"release:{release.version} AND event.timestamp:>{timezone.now()}"
+                        )
+                        logger.info(
+                            "Attempt subscribe project to activated alert rule",
+                            extra={
+                                "trigger": trigger,
+                                "query_extra": query_extra,
+                            },
+                        )
```

Comment:
thoughts on moving the log into this function? that way we will log it every time the function is invoked

Reply:
I suppose i can make this logger more general 🤔